src/services/docGenService.ts

```diff
@@ -228,6 +222,36 @@ export class DocGenService {
     );
   }
 
+  private getEventByProject(
+    eventMap: Map<string, ManifestCacheProjectAddedEvent>,
+  ): ManifestCacheProjectAddedEvent | undefined {
+    if (window.activeTextEditor === undefined || eventMap === undefined) {
+      return;
+    }
+
+    const currentFilePath = window.activeTextEditor.document.uri;
+    this.dbtTerminal.debug(
+      "getting event for project, currentFilePath: ",
+      currentFilePath.fsPath,
+    );
+    const projectRootpath =
+      this.dbtProjectContainer.getProjectRootpath(currentFilePath);
+    if (projectRootpath === undefined) {
+      this.dbtTerminal.debug(
+        "no project for currentFilePath: ",
+        currentFilePath.fsPath,
+      );
+      return;
+    }
+
+    const event = eventMap.get(projectRootpath.fsPath);
+    if (event === undefined) {
+      this.dbtTerminal.debug("no event for project: ", projectRootpath.fsPath);
+      return;
+    }
+    return event;
+  }
```

Comment:
Good candidate to move to dbtProjectService.

Reply:
moved